backend/substrapp/tasks/tasks.py

```diff
@@ -626,27 +630,93 @@ def do_task(subtuple, tuple_type):
 
 def _do_task(client, subtuple_directory, tuple_type, subtuple, compute_plan_id, rank, org_name, compute_plan_tag):
 
-    algo_hash = subtuple['algo']['hash']
-    model_folder = '/sandbox/model'
-    model_path = path.join(subtuple_directory, 'model')
-    data_path = path.join(subtuple_directory, 'data')
-    pred_path = path.join(subtuple_directory, 'pred')
-    opener_file = path.join(subtuple_directory, 'opener/opener.py')
-    algo_path = path.join(subtuple_directory)
-    algo_docker = get_algo_image_name(algo_hash)
-    algo_docker_name = f'{tuple_type}_{subtuple["key"][0:8]}'
-    output_head_model_filename = 'head_model'
-    output_trunk_model_filename = 'trunk_model'
+    common_volumes, compute_volumes = prepare_volumes(
+        client, subtuple_directory, tuple_type, compute_plan_id, compute_plan_tag)
+
+    # Add node index to environment variable for the compute
+    node_index = os.getenv('NODE_INDEX')
+    if node_index:
+        environment = {'NODE_INDEX': node_index}
+    else:
+        environment = {}
+
+    container_name, command = generate_container_name_and_cmd(tuple_type, subtuple, rank)
```

Comment:
Nit, and maybe out of scope: what about simplifying the container name (there's only one operation - train, predict, etc - we can do on each type, no need to specify it?) 

```suggestion
    command = get_command(tuple_type, subtuple, rank)
    container_name = f'{tuple_type}_{subtuple["key"][0:8]}'
```

Reply:
Done